packages/venia-concept/src/actions/cart/asyncActions.js

```diff
@@ -107,6 +107,13 @@ export const addItemToCart = (payload = {}) => {
             );
 
             dispatch(actions.addItem.receive({ cartItem, item, quantity }));
+
+            // 2019-02-07  Moved these dispatches to the success clause of
+            // addItemToCart. The cart should only open on success.
+            // In the catch clause, this action creator calls its own thunk,
+            // so a successful retry will wind up here anyway.
```

Comment:
That would be ideal, and as soon as we have a loading mask for the cart, we should go back to that. But when we don't have a loading mask, the experience is:

- I click "Add to cart"
- The cart slides open saying I have no items
- A second goes by
- Smash cut to cart items

It's not a good experience. Since right now we have to choose between that and a delay in opening the cart, I think we have to do the latter.

Reply:
Right, in either case the user is left wondering what's happening while the request is pending, which isn't a good experience. The reason I didn't want to adjust actions is that the goal was always to add a loading state and mask to the cart.

We now have a loading state and mask in the cart, thanks to #732 and #892, but we're going to have to find these action changes and undo them in order to take advantage of them. Not a huge deal, but kind of a waste.